server/db/level/utils.js

```diff
@@ -0,0 +1,35 @@
+const __ = require('config').universalPath
+const { forceArray } = __.require('builders', 'utils')
+const { Promise } = __.require('lib', 'promises')
+
+const streamPromise = stream => new Promise((resolve, reject) => {
+  const results = []
+  return stream
+  .on('data', results.push.bind(results))
+  .on('end', () => resolve(results))
+  .on('error', reject)
+})
+
+module.exports = {
+  streamPromise,
+
+  reset: db => {
+    return streamPromise(db.createKeyStream())
+    .then(keys => {
+      const ops = keys.map(key => ({ type: 'del', key }))
+      return db.batch(ops)
+    })
+  },
+
+  formatBatchOps: ops => forceArray(ops).map(setDefaultType),
+
+  // Levelup rejects `null` or `undefined` values, so the empty value should
+  // be an empty string.
+  // Useful when the key alone stores all the data that needs to be stored
+  emptyValue: ''
```

Comment:
so store an empty string in a utils file to serve it in other files like `debounce_email` below? why not using directly an empty string there instead of defining one here ?

Reply:
For the sack of documentation: it looks more explicit to me to have an `emptyValue` variable with a comment explaining why we use that, rather than just use an obscure `''` in places where it's needed. Unless you add a comment in all the places that pattern is used. But then, why not having it in one place

The same idea is used in [`lib/couch` `minKey` and `maxKey`](https://github.com/inventaire/inventaire/blob/c78e62cd/server/lib/couch.js#L26-L31).

It makes a lot of sense to me to have those values in those lib files, as those values are very much bound to respectively levelup and couchdb internals